tests/dialects/test_postgres.py

```diff
@@ -724,6 +724,28 @@ def test_postgres(self):
         self.validate_identity("cast(a as FLOAT8)", "CAST(a AS DOUBLE PRECISION)")
         self.validate_identity("cast(a as FLOAT4)", "CAST(a AS REAL)")
 
+        self.validate_all(
+            "1 / DIV(4, 2)",
+            read={
+                "postgres": "1 / DIV(4, 2)",
+            },
+            write={
+                "sqlite": "1 / CAST(CAST(CAST(4 AS REAL) / 2 AS INTEGER) AS REAL)",
```

Comment:
The `CAST(4 AS REAL)` here looks a bit weird, I don't think it should've been added. The root cause is that the `IntDiv` generator [doesn't properly populate](https://github.com/tobymao/sqlglot/blob/main/sqlglot/generator.py#L3136-L3142) the `typed` and `safe` args, disregarding the dialect's semantics. It seems like that method was added [a long time ago](https://github.com/tobymao/sqlglot/commit/193d4f2c0463c23aba220c7c5243a83d248078d9), so this is most likely like an oversight.

Reply:
> I don't think it should've been added

Reasoning is that Postgres uses typed division, so since we're generating a `Div` for the `IntDiv`, the typed division semantics should be respected. See for example:

```python
>>> import sqlglot
>>> sqlglot.transpile("1/2", "postgres", "sqlite")
['1 / 2']
```

Reply:
May be best to address this in a followup PR after all. I played around a bit and it seems like we'll need to think more about how to parse / generate `IntDiv`.